### ISSUE TITLE
i'm gettting unsaved warning even while saving the form

### ISSUE THREAD

**Issue by shreyas-jadhav** [marmelab/react-admin#6966]
**What you were expecting:**
show warning only when user tries to navigate and not while clicking the save button. 

**What happened instead:**
when i click save button still i get the unsaved warning. 

https://user-images.githubusercontent.com/73429007/145399140-9765e202-2286-4c82-b79c-c22760055415.mp4

**Steps to reproduce:**
add `<SimpleForm warnWhenUnsavedChanges>`

**Related code:**
this is how my basic edit component look like. i don't think anything wrong here, but i don't know if there is.
```tsx
export const AudioTrackEdit = (props: EditProps): JSX.Element => {
  return (
    <Edit {...props}>
      <SimpleForm warnWhenUnsavedChanges>
        <RangeSlider
          source="panpos"
          label="Pan Position"
          min={-1}
          max={1}
          step={0.1}
          forceMax={1}
        />
        <RangeSlider
          source="panduration"
          label="Pan Duration (seconds)"
          step={0.1}
        />
        <NumberInput source="banned_duration" defaultValue={0} />
        <BooleanInput source="repeatrecordings" label="Repeat Recordings" />
        <BooleanInput source="start_with_silence" label="Start With Silence" />
        <BooleanInput
          source="fadeout_when_filtered"
          label="Fade Out When Filtered"
        />
      </SimpleForm>
    </Edit>
  );
};

```
full code at
https://github.com/shreyas-jadhav/roundware-react-admin/blob/294be8eae16ba916c0929e7a7069c8e5ddb42be9/src/components/AudioTrack.tsx#L50

**Environment**

* React-admin version: 3.19.2
* React version: 17.0.2
* Browser: firefox

**Comment by WiXSL** [comment]
A problem like this has been fixed in the latest version (3.19.3).
Please, try it and see if It's solved.

**Comment by shreyas-jadhav** [comment]
i updated to 3.19.3 and it seems to be fixed. 
thank you.
